### PR TITLE
Fix missing ocean tiles

### LINKED ISSUE #11975: Draping artifacts on ocean tiles
With globe or terrain enabled, trianglular artifacts appear in specific locations between zoom levels 5 and six. This occurs with the tile 5/27/14.

 1.  Go to http://localhost:9966/debug/projections.html#5.5/19/125
 2. Enable globe OR terrain.

image

Acceptance Criteria:

These artifacts should no longer appear and be verified with a render test.

### PR DESCRIPTION
**Pull request by avpeery**
Fixes #11975. Similar logic to PR #12203, but epsilon value is only applied when globe projection and to the left bound (see definition for mat4.ortho [here](https://glmatrix.net/docs/module-mat4.html#ortho))

Test with [codesandbox](https://codesandbox.io/s/young-feather-bphr2q?file=/index.html) on different devices, many thanks!
- Already tested on Macbook Pro Monterey 12.6 (chrome, safari, firefox), and iPhone 12 Pro (Safari)
## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [x] write tests for all new functionality
 - [x] manually test the debug page
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Fix missing ocean tiles when globe is enabled.</changelog>`

### CODE REVIEW

**Comment by karimnaaji** [comment]
@avpeery I haven't been able to reproduce from my machine with this new epsilon value. Before moving this forward, it would be worth doing two things:
- Test that more on different devices (with varying desktop/os/mobile...) due to chance of regression as we noticed in https://github.com/mapbox/mapbox-gl-js/pull/12203#issuecomment-1233307545
- Reintegrate the render tests from https://github.com/mapbox/mapbox-gl-js/pull/12203/files

File: src/terrain/terrain.js [hunk 1]
```diff
@@ -260,7 +260,8 @@ export class Terrain extends Elevation {
         this._sourceTilesOverlap = {};
         this.proxySourceCache = new ProxySourceCache(style.map);
         this.orthoMatrix = mat4.create();
-        mat4.ortho(this.orthoMatrix, 0, EXTENT, 0, EXTENT, 0, 1);
+        const epsilon = this.painter.transform.projection.name === 'globe' ?  .015 : 0; // Experimentally the smallest value to avoid rendering artifacts (https://github.com/mapbox/mapbox-gl-js/issues/11975)
```
Comment (SnailBones) [inline src/terrain/terrain.js:263]: Do we also need to check for terrain here?

Comment (avpeery) [reply src/terrain/terrain.js:263]: I'm not seeing the missing tile with terrain (no globe) with this fix, but will test out on various devices! I also hit a [failed render test](https://output.circle-artifacts.com/output/job/ae72b852-170d-4096-b020-ab2f4e1a6bd1/artifacts/0/test/integration/render-tests/index.html) when I removed the globe check (no significant difference though)

Comment (avpeery) [reply src/terrain/terrain.js:263]: Here's a [codesandbox](https://codesandbox.io/s/young-feather-bphr2q?file=/index.html) with this change

**Review by karimnaaji — APPROVED** [review]
LGTM. Tested on both chrome iPhone se and macOS.

**Review by SnailBones — APPROVED** [review]
Looks good to me! Tested on Linux, Windows and Android.

**Pull request merged**